### PR TITLE
refactor: use web3-events for blockchain listening

### PR DESCRIPTION
**Pull request by AuHau**
Sooooo I managed to pull it to the finish line, hopefully. I don't have running the dev setup for the latest contracts so I did not actually run it, maybe it would good idea if somebody would run it prior to merging. I was mainly rallying on the tests that @nduchak wrote (Thanks btw!). Especially the integration tests were really good for me! Although as the RNS part is not covered, I am not completely sure about it working.

Anyhow brief summary of the changes:
 - moved all of the blockchain listening logic to `@rsksmart/web3-events` package
 - refactored the DB backup logic
 - did some general moving and tweaking

### CODE REVIEW

**Comment by nduchak** [comment]
Nice. You did a great job 👏

File: src/store.ts [hunk 1]
```diff
@@ -5,27 +5,27 @@ import type { Sequelize } from 'sequelize'
...
+    'web3events.rns.reverse.lastFetchedBlockNumber': 'int',
+    'web3events.rns.reverse.lastFetchedBlockHash': 'string',
+    'web3events.rns.placement.lastFetchedBlockNumber': 'int',
+    'web3events.rns.placement.lastFetchedBlockHash': 'string',
+    'web3events.rns.owner.lastProcessedBlockNumber': 'int',
+    'web3events.rns.owner.lastProcessedBlockHash': 'string',
+    'web3events.rns.reverse.lastProcessedBlockNumber': 'int',
+    'web3events.rns.reverse.lastProcessedBlockHash': 'string',
+    'web3events.rns.placement.lastProcessedBlockNumber': 'int',
+    'web3events.rns.placement.lastProcessedBlockHash': 'string'
```
Comment (jurajpiar) [inline src/store.ts:28]: could be DRYer

File: package.json [hunk 2]
```diff
@@ -81,13 +81,13 @@
     "peer-id": "^0.14.2",
     "reflect-metadata": "^0.1.13",
     "rlp": "^2.2.6",
-    "sequelize": "^5.21.10",
+    "rxjs": "^6.6.3",
```
Comment (jurajpiar) [inline package.json:84]: haha I think I suggested Observable  back in March/April

**Review by jurajpiar — APPROVED** [review]
LG

File: src/app.ts [hunk 3]
```diff
@@ -49,18 +61,32 @@ export async function appFactory (options: AppOptions): Promise<{ app: Applicati
...
 
-  // Authenticatoin service
+  // Authentication service
   app.configure(authentication)
 
   // Custom general services
   app.configure(sequelize)
   app.configure(configureStore)
-  app.configure(blockchain)
+  await waitForConfigure(app, blockchain)
```
Comment (nduchak) [inline src/app.ts:70]: hmm... as I understand it depends on the order

File: src/blockchain/index.ts [hunk 4]
```diff
@@ -19,6 +18,15 @@ export function ethFactory (): Eth {
   return new Eth(provider)
 }
 
+export async function web3eventsFactory (eth: Eth, sequelize: Sequelize): Promise<Web3Events> {
```
Comment (nduchak) [inline src/blockchain/index.ts:21]: Amazing 🥇

**Review by nduchak — APPROVED** [review]
LGTM

**Comment by nduchak** [comment]
Now, just need to rebase it 😈

File: src/store.ts (hunk 1 above)
Comment (AuHau) [reply src/store.ts:28]: Good idea!

File: package.json (hunk 2 above)
Comment (AuHau) [reply package.json:84]: Hmm don't recall, but it is here only for the tracking progress at https://www.npmjs.com/package/listr library

File: src/app.ts (hunk 3 above)
Comment (AuHau) [reply src/app.ts:70]: Yeah, it is not ideal, but the Web3Events needs to be initialized (its models have to be added to Sequelize and synced) before we can start initialization of the services...

**Pull request merged**